homeworks/zsoltcsengeri/6_testing/test_bankaccount.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+from bankaccount import BankAccount
+
+"""
+Write at least 2 fixtures (2 different BankAccount objects) that you use in your tests.
+Write at least one test that uses @pytest.mark.parametrize 
+to run the test with multiple inputs (e.g., test the deposit() 
+method with exactly 0 and negative number inputs)
+"""
+
+
+# Fixture for bankaccount_1 account
+@pytest.fixture
+def bankaccount_1():
+    # A BankAccount for Zsolt with $100 balance
+    return BankAccount("Zsolt", 100)
+
+
+# Fixture for bankaccount_2 account
+@pytest.fixture
+def bankaccount_2():
+    # A BankAccount for Maryna with $50 balance
+    return BankAccount("Maryna", 50)
+
+
+# Test for bankaccount_1 deposit
+def test_deposit_bankaccount_1(bankaccount_1):
+    bankaccount_1.deposit(600)
+    assert bankaccount_1.balance == 700
+
+
+# Test for bankaccount_2 deposit
+def test_deposit_bankaccount_2(bankaccount_2):
+    bankaccount_2.deposit(200)
+    assert bankaccount_2.balance == 250
+
+
+# Parametrized test for invalid deposit inputs for bankaccount_1
+@pytest.mark.parametrize(
+    "amount, expected_exception",
+    [
+        (-100, ValueError),  # Negative deposit
+        (0, ValueError),  # Zero deposit
+    ],
+)
+def test_deposit_invalid_input(bankaccount_1, amount, expected_exception):
+    # Test invalid deposit for bankaccount_1
+    with pytest.raises(expected_exception):
+        bankaccount_1.deposit(amount)
+
+
+# Parametrized test for invalid deposit inputs for bankaccount_2
+@pytest.mark.parametrize(
+    "amount, expected_exception",
+    [
+        (-100, ValueError),  # Negative deposit
+        (0, ValueError),  # Zero deposit
+    ],
+)
+def test_deposit_invalid_input(bankaccount_2, amount, expected_exception):
+    # Test invalid deposit for bankaccount_2
+    with pytest.raises(expected_exception):
+        bankaccount_2.deposit(amount)
+
+    """
+    Edge Case Test:
+ 
+    """
+
+
+# Transferring money to non-BankAccount objects
+@pytest.mark.parametrize(
```

Comment:
Itt szuperül megadod a parametrize-ban a paramétereket amikre le kéne fusson a teszt külön külön, viszont végül nem használod a tesztben egyiket sem.
![image](https://github.com/user-attachments/assets/dae57839-533a-48a3-8838-067d0aa0d5f7)
Ezen a screenshoton bekarikáztam pirossal az invalid_account paramétert. Látszik, hogy az IDE is jelzi a paraméter színével hogy nem használod sehol ezt az értéket. Ennek a paraméternek a helyére kerülne be az aktuális érték a parametrize listából (először a None, majd a 123, a [] és a {}). 
a 83. sorban tehát a bankaccount_1.transfer()-nek ezt az invalid_account paramétert kéne átadni, így:
`bankaccount_1.transfer(invalid_account)`
Most is lefut 4-szer a teszted ahogy várod, mert a parametrize-ban 4 teszt esetet adsz meg, azonban minden esetben erre a list of dictionary-re fog lefutni a teszt: `[{1, 2, 3, 4, 5}]` mert ezt hard-code-oltad bele a tesztbe.
Mivel TypeError-t vársz mindenhol, még egyszerűbb is lehet a teszted egyébként:

```
@pytest.mark.parametrize(
    "invalid_account",
    [None, 123, [], {}], 
)
def test_transfer_to_invalid_target(invalid_account, bankaccount_1):
    with pytest.raises(TypeError):
        bankaccount_1.transfer(invalid_account, 50)
```

Itt az történik hogy a parametrize listában csak az input értékeket adom meg, ezek bekerülnek minden hívásnál az invalid_account helyére.Mivel a pytest.raises()-ben megadod hogy TypeError-t vársz, így minden teszt esetet erre fog tesztelni.
Át is neveztem a tesztedet transferre, mert alapból deposit-nak hívtad, de a transfer metódust teszteljük. Remélem ez segít kicsit tisztázni a dolgokat, ha van bármi kérdés jelezd kérlek.

Reply:
Szia István,

Köszönöm az iránymutatást, ismét sokat tanultam belőle.
Elvégeztem a kért változtatásokat, és commitoltam az iránymutatásod alapján.

Üdv,
Zsolt